pkg/resource/statefulset.go

```diff
@@ -245,14 +263,32 @@ func (b StatefulSetBuilder) MakeContainers() []corev1.Container {
 			},
 		},
 	}
+
+	if utilfeature.DefaultMutableFeatureGate.Enabled(features.RunAsNonRoot) {
+		// Setup the pod to not run as root
+		container[0].SecurityContext = &corev1.SecurityContext{
+			RunAsUser:    userId,
+			RunAsGroup:   userId,
+			RunAsNonRoot: ptr.Bool(true),
+		}
+
+		// we need this value because K8s does not allow for the setting
+		// of chown perms on a secret
+		container[0].Env = append(container[0].Env,
+			corev1.EnvVar{
+				Name:  "COCKROACH_SKIP_KEY_PERMISSION_CHECK",
+				Value: "true",
+			})
+	}
+
+	return container
 }
 
 func (b StatefulSetBuilder) secureMode() string {
 	if b.Spec().TLSEnabled {
-		return " --certs-dir=/cockroach/cockroach-certs/"
+		return "--certs-dir=/cockroach/cockroach-certs/"
```

Comment:
This is the biggest bit that's jumping out at me 🤔  as I don't see any accompanying change?
Or was there just an additional space there in the first place?

Reply:
hrmmmm

Reply:
I do not think that is called anywhere.  I am trying to get this working in debug mode.

Reply:
It is not called anywhere according to grep.